code/Test_Definitions/NumberVerification_verify.feature

```diff
@@ -0,0 +1,137 @@
+    And the response property "$.devicePhoneNumberVerified" is true
+
+
+  @NumberVerification_verify101_match_false
+  Scenario:  verify phone number NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER1, network connection and access token matches NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER2
+    Given they use the base url over a mobile connection
+    And the resource is "/verify"
+    And they acquired a valid access token associated with NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER1 through OIDC authorization code flow or CIBA
+    And one of the scopes associated with the access token is number-verification:verify
+    When the HTTPS "POST" request is sent
+    And the mobile connection is associated with NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER1
+    And the request body has the field phoneNumber with a value of NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER2
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response header "Content-Type" is "application/json"
+    And the response body complies with the OAS schema at "/components/schemas/SendCodeResponse"
+    Then the response status code is 200
+    And the response property "$.devicePhoneNumberVerified" is false
+
+  @NumberVerification_verify200_missing_phone_number_in_request
+  Scenario:  verify phone number but no phonenumber in request
+    Given they use the base url over a mobile connection
+    And the resource is "/verify"
+    And they acquired a valid access token associated with NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER1 through OIDC authorization code flow or CIBA
+    And one of the scopes associated with the access token is number-verification:verify
+    When the HTTPS "POST" request is sent
+    And the mobile connection is associated with NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER1
+    And the request body has NO the field phoneNumber
+    And the response header "x-correlator" has same value as the request header "x-correlator"
+    And the response header "Content-Type" is "application/json"
+    And the response body complies with the OAS schema at "/components/schemas/SendCodeResponse"
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+  @NumberVerification_verify201_missing_scope
+  Scenario:  verify phone number with valid access token but scope number-verification:verify is missing
+    Given they use the base url over a mobile connection
+    And the resource is "/verify"
+    And they acquired a valid access token associated with NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER1 through OIDC authorization code flow or CIBA
+    And none of the scopes associated with the access token is number-verification:verify
+    When the HTTPS "POST" request is sent
+    And the mobile connection is associated with NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER1
+    And the request body has NO the field phoneNumber
```

Comment:
If the phoneNumber is present we expect same outcome right? in this case probably I suggest to have it valued. It will avoid confusion with 2 causes of failure.

Reply:
The access token should always be checked first, so the rest of the request parameter do not really matter.
But I changed that.